android/app/src/main/java/com/cBreez/client/BreezSdkWorker.kt

```diff
@@ -0,0 +1,160 @@
+package com.cBreez.client
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.ForegroundInfo
+import androidx.work.WorkerParameters
+import breez_sdk.BlockingBreezServices
+import breez_sdk.BreezEvent
+import breez_sdk.EnvironmentType
+import breez_sdk.EventListener
+import breez_sdk.GreenlightNodeConfig
+import breez_sdk.NodeConfig
+import breez_sdk.Payment
+import breez_sdk.PaymentDetails
+import breez_sdk.PaymentStatus
+import breez_sdk.connect
+import breez_sdk.defaultConfig
+import breez_sdk.mnemonicToSeed
+import com.cBreez.client.Constants.NOTIFICATION_ID_PAYMENT_RECEIVED
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+
+// SDK events listener
+class SDKListener(private val paymentListener: (payment: Payment) -> Void) : EventListener {
+    val TAG = "SDKListener"
+
+    override fun onEvent(e: BreezEvent) {
+        Log.v(TAG, "Received event $e")
+        if (e is BreezEvent.InvoicePaid) {
+            Log.v(
+                TAG,
+                "Received payment. Bolt11: ${e.details.bolt11}\nPayment Hash:${e.details.paymentHash}"
+            )
+            e.details.payment?.let {
+                paymentListener(it)
+            }
+        }
+    }
+}
+
+class BreezSdkWorker(appContext: Context, workerParams: WorkerParameters) :
+    CoroutineWorker(appContext, workerParams) {
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        return ForegroundInfo(
+            NOTIFICATION_ID_PAYMENT_RECEIVED,
+            createNotification(
+                "Incoming payment",
+                NotificationManager.IMPORTANCE_LOW,
+                true
+            )
+        )
+    }
+
+    override suspend fun doWork(): Result = coroutineScope {
+        // TODO(_): Improve error handling
+        val breezSDK = connectSDK(::paymentListener)
+        val paymentHash =
+            inputData.getString("PAYMENT_HASH") ?: return@coroutineScope Result.failure()
+        val payment = breezSDK?.paymentByHash(paymentHash)
```

Comment:
We need to poll here like we do in iOS right?

Reply:
Done as part of [BreezSdkWorker second draft](https://github.com/breez/c-breez/pull/737/commits/253c8520b683d788c875ba66709a8b55f592fc57)